sebs/aws/triggers.py

```diff
@@ -123,3 +126,171 @@ def serialize(self) -> dict:
     @staticmethod
     def deserialize(obj: dict) -> Trigger:
         return HTTPTrigger(obj["url"], obj["api-id"])
+
+
+class QueueTrigger(Trigger):
+    def __init__(self, fname: str, deployment_client: Optional[AWS] = None):
+        super().__init__()
+        self.name = fname
+        self._deployment_client = deployment_client
+
+    @staticmethod
+    def typename() -> str:
+        return "AWS.QueueTrigger"
+
+    @property
+    def deployment_client(self) -> AWS:
+        assert self._deployment_client
+        return self._deployment_client
+
+    @deployment_client.setter
+    def deployment_client(self, deployment_client: AWS):
+        self._deployment_client = deployment_client
+
+    @staticmethod
+    def trigger_type() -> Trigger.TriggerType:
+        return Trigger.TriggerType.QUEUE
+
+    def sync_invoke(self, payload: dict) -> ExecutionResult:
+
+        self.logging.debug(f"Invoke function {self.name}")
+
+        # Init clients
+        lambda_client = self.deployment_client.get_lambda_client()
+        sqs_client = boto3.client('sqs', region_name=self.deployment_client.config.region)
+
+        serialized_payload = json.dumps(payload)
+
+        # Create queue
+        self.logging.debug(f"Creating queue {self.name}")
+
+        queue_url = sqs_client.create_queue(QueueName=self.name)["QueueUrl"]
+        queue_arn = sqs_client.get_queue_attributes(
+                                                    QueueUrl=queue_url,
+                                                    AttributeNames=["QueueArn"]
+                                                   )["Attributes"]["QueueArn"]
+
+        self.logging.debug("Created queue")
+
+        # Add queue trigger
+        if (not len(lambda_client.list_event_source_mappings(EventSourceArn=queue_arn,
+                                                             FunctionName=self.name)
+                                                        ["EventSourceMappings"])):
+            lambda_client.create_event_source_mapping(
+                EventSourceArn=queue_arn,
+                FunctionName=self.name,
+                MaximumBatchingWindowInSeconds=1
+            )
+
+        # Publish payload to queue
+        sqs_client.send_message(QueueUrl=queue_url, MessageBody=serialized_payload)
+        self.logging.info(f"Sent message to queue {self.name}")
+
+        # TODO(oana): gather metrics
+
+    def async_invoke(self, payload: dict) -> concurrent.futures.Future:
+
+        pool = concurrent.futures.ThreadPoolExecutor()
+        fut = pool.submit(self.sync_invoke, payload)
+        return fut
+
+    def serialize(self) -> dict:
+        return {"type": "Queue", "name": self.name}
```

Comment:
When serializing, it would be good to store URL/ARN - when we retrieve data from cache, we will be able to invoke without any additional queries through AWS SDK.